### ISSUE TITLE
Trim in .net 6 breaks self contained single file publish mode

### ISSUE THREAD

**Issue by hahn-kev** [AvaloniaUI/Avalonia#6892]
**Describe the bug**
References required for file dialogs will break. There's an exception when trying to use them.

**To Reproduce**

1. Run an avalonia app with a folder dialog
2. publish it with .net 6 with trim enabled
3. see the exception

**Desktop (please complete the following information):**

- OS: Windows
- Version 0.10.8

**Additional context**
Work around, add these settings into your project file
```
<TrimMode>copyused</TrimMode>
<BuiltInComInteropSupport>true</BuiltInComInteropSupport>
```
this will prevent the needed files from being trimmed, and prevent com from being disabled which is needed.

**Comment by luthfiampas** [comment]
I use the latest version (0.10.10). Actually, this problem occurs with `OpenFileDialog` as well. In Linux (Debian), IL trimming doesn't break open folder/file dialog.

**Comment by Mikolaytis** [comment]
I think we can create a PR adding a trim options to some types/assemblies?
https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options

**Comment by maxkatz6** [comment]
@luthfiampas problem with dialog was fixed (not yet released). But drag-n-drop might have the same problem.

**Comment by hez2010** [comment]
I've tested and can confirm that drag-n-drop has the same problem.

**Comment by maxkatz6** [comment]
There are more places:
```
Exception Info: System.TypeLoadException: Could not load type 'System.StubHelpers.InterfaceMarshaler' from assembly 'System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'.
   at System.Runtime.InteropServices.Marshal.GetDelegateForFunctionPointerInternal(IntPtr , Type )
   at System.Runtime.InteropServices.Marshal.GetDelegateForFunctionPointer[TDelegate](IntPtr )
   at System.Runtime.InteropServices.FunctionWrapper.<>c__DisplayClass0_0`1.<Load>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode )
   at System.Lazy`1.ExecutionAndPublication(LazyHelper , Boolean )
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at System.Runtime.InteropServices.FunctionWrapper`1.get_Delegate()
   at System.Drawing.SafeNativeMethods.Gdip.GdipCreateBitmapFromStream(IStream stream, IntPtr& bitmap)
   at System.Drawing.Bitmap..ctor(Stream stream, Boolean useIcm)
   at System.Drawing.Bitmap..ctor(Stream stream)
   at Avalonia.Win32.CursorFactory.LoadSystemDrawingBitmap(IBitmapImpl bitmap) in C:\Work\Projects\Avalonia\src\Windows\Avalonia.Win32\CursorFactory.cs:line 114
```

**Comment by colejohnson66** [comment]
Is this fixed in 0.10.17?

**Comment by hez2010** [comment]
You can workaround it by preserving System.StubHelpers.InterfaceMarshaler

**Comment by KieranDevvs** [comment]
Is this fixed? The avalonia templates in VS link to this issue but this is marked as completed? I've tried to publish a self contained trimmed (link) executable but it crashes upon start. Im assuming its a case that the work has been done but just not released?

**Comment by maxkatz6** [comment]
It was fixed in 11.0 preview.
